apps/lib/s_cb.c

```diff
@@ -1574,3 +1574,16 @@ void ssl_print_secure_renegotiation_notes(BIO *bio, SSL *s)
         BIO_printf(bio, "This TLS version forbids renegotiation.\n");
     }
 }
+
+int progress_cb(EVP_PKEY_CTX *ctx)
+{
+    BIO *b = EVP_PKEY_CTX_get_app_data(ctx);
+    int p = EVP_PKEY_CTX_get_keygen_info(ctx, 0);
+    static const char symbols[] = ".+*\n";
+    char c = (p >= 0 && (size_t)p < sizeof(symbols) - 1) ? symbols[p] : '?';
```

Comment:
There seems to be an off by 1 here.

Reply:
> There seems to be an off by 1 here.

I didn't change this code, just moved it around.  In any case, it's correct.  The string is 5 bytes long, including the terminating NUL.  So `sizeof(symbols) - 1` would be `4`.  But C arrays are zero-indexed, so the valid range is `0..3` (4 values).

Reply:
You're right, I got sizeof() includes the NUL, strlen() doesn't.